internal/model/user_server.go

```diff
@@ -160,10 +160,22 @@ func (this *ServerUser) GetCourseRole(courseID string) CourseUserRole {
 }
 
 // Convert this server user into a course user for the specific course.
+// Set courseOnly to false if high level server users should be converted to a course user.
 // Will return (nil, nil) if the user is not enrolled in the given course.
-func (this *ServerUser) ToCourseUser(courseID string) (*CourseUser, error) {
+func (this *ServerUser) ToCourseUser(courseID string, courseOnly bool) (*CourseUser, error) {
```

Comment:
The idea of adding in a parameter here is good, but I think it should be inverted.
I think that `courseOnly` is not intuitive,
and it should be inverted into `escalateServerAdmin` or something like that.

Changing the name can also hint in a change of semantics where we can escalate server admins who are enrolled in a course to course owners.
This would change it so that escalation only happens on request.